ipld/graphsync/src/libp2p/behaviour.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::config::GraphSyncConfig;
+use super::handler::GraphSyncHandler;
+use crate::{Extensions, GraphSyncMessage};
+use cid::Cid;
+use forest_ipld::selector::Selector;
+use futures::task::Context;
+use futures_util::task::Poll;
+use libp2p::core::connection::ConnectionId;
+use libp2p::swarm::{
+    protocols_handler::ProtocolsHandler, NetworkBehaviour, NetworkBehaviourAction, NotifyHandler,
+    PollParameters,
+};
+use libp2p::{Multiaddr, PeerId};
+use log::debug;
+use std::collections::{HashSet, VecDeque};
+
+/// The GraphSync behaviour that gets consumed by the Swarm.
+#[derive(Default)]
+pub struct GraphSync {
+    /// Config options for the service
+    config: GraphSyncConfig,
+
+    /// Queue of events to processed.
+    events: VecDeque<NetworkBehaviourAction<GraphSyncMessage, ()>>,
+
+    // TODO just temporary, will probably have to attach some data with peers
+    peers: HashSet<PeerId>,
+}
+
+impl GraphSync {
+    /// Creates a new GraphSync behaviour
+    pub fn new(config: GraphSyncConfig) -> Self {
+        Self {
+            config,
+            ..Default::default()
+        }
+    }
+
+    /// Initiates GraphSync request to peer given root and selector.
+    pub fn send_request(
+        &mut self,
+        _peer_id: PeerId,
+        _root: Cid,
+        _selector: Selector,
+        _extensions: Extensions,
+    ) {
+        todo!()
+        // self.events
+        //     .push_back(NetworkBehaviourAction::NotifyHandler {
+        //         peer_id,
+        //         // TODO once request manager logic built out
+        //         event: todo!(),
+        //         handler: NotifyHandler::Any,
+        //     });
```

Comment:
Maybe remove for now?

Reply:
yeah was just leaving for convenience, I'll remove for now tho